docker_control.sh

```diff
@@ -1,49 +1,63 @@
 #!/bin/bash
 set -e
 
+PROJROOT=$(git rev-parse --show-toplevel)
+export PROJROOT
+cd "${PROJROOT}"
+
 export IMAGE_NAME='starryoss/sysadmin-build'
-export NEW_TAG=":${2}"
+
+if [ ! -z "${2}" ]; then
+  export NEW_TAG=":${2}"
+else
+  export NEW_TAG=":latest"
+fi
 
 show_help() {
 cat << EOF
-Usage: ${0##*/} [-dbtish*]
+Usage: ${0##*/} [-dbtisch*]
  -h          display this help and exit
  -d          build dependency image
  -b          build the user image from the dependency image
  -t          build and test. Non-interactive
  -i          run container interactively
  -s          start local sysadmin
+ -c          clean Sysadmin tmp and history files
 EOF
 }
 
-
 NET_OPTS="--name sysadmin --network sysadmin -p 4000:4000"
-VOL_OPTS="-v `pwd`:/home/user/sysadmin --workdir /home/user/sysadmin"
+VOL_OPTS="-v ${PROJROOT}:/home/user/sysadmin --workdir /home/user/sysadmin"
 
 # try cleaning up sysadmin network. It fails if a container
 # is using it and that is okay
 docker network rm sysadmin > /dev/null 2>&1 || true
-# make network for sysadmin
+
+# make docker network for sysadmin
 docker network inspect sysadmin > /dev/null 2>&1 || \
   docker network create --attachable  sysadmin
 
-while getopts ":dbtish*:" opt; do
+function clean_state {
+  rm -rfv historyTmp tmp
+}
+
+while getopts ":dbtisch*:" opt; do
   case $opt in
     d)  docker build -t "${IMAGE_NAME}${NEW_TAG}" docker_builds/
         exit $?
       ;;
 
     b)  docker pull "${IMAGE_NAME}"  # pull Dockerhub image
         docker build --cache-from="${IMAGE_NAME}" -t "${IMAGE_NAME}:latest" docker_builds/  # build using DH image as cache
-        docker build --build-arg=USERID=`id -u` . -t sysadmin_tester
+        docker build --build-arg=USERID="$(id -u)" . -t sysadmin_tester
         exit $?
       ;;
     t)  docker run -t --rm ${VOL_OPTS} ${NET_OPTS} -u user sysadmin_tester \
-          /bin/bash -c \
+          /bin/bash -i -c \
           "rm -rf /home/user/sysadmin/build && \
            mkdir -p /home/user/sysadmin/build && \
            cd /home/user/sysadmin/build && \
-           cmake .. && make check && make "
+           cmake .. && make check -j8 && make -j8"
```

Comment:
-j8 should probably be replaced with the number of CPUs available - this will make builds slower on machines with fewer cores.

https://stackoverflow.com/questions/6481005/how-to-obtain-the-number-of-cpus-cores-in-linux-from-the-command-line